mne/annotations.py

```diff
@@ -741,49 +798,20 @@ def events_from_annotations(raw, event_id=None, regexp=None, use_rounding=True,
 
     annotations = raw.annotations
 
-    inds = raw.time_as_index(annotations.onset, use_rounding=use_rounding,
-                             origin=annotations.orig_time) + raw.first_samp
-
-    # Filter out the annotations that do not match regexp
-    regexp_comp = re.compile('.*' if regexp is None else regexp)
-
-    if event_id is None:
-        event_id = Counter()
-
-    event_id_ = dict()
-    dropped = []
-    for desc in annotations.description:
-        if desc in event_id_:
-            continue
-
-        if regexp_comp.match(desc) is None:
-            continue
-
-        if isinstance(event_id, dict):
-            if desc in event_id:
-                event_id_[desc] = event_id[desc]
-            else:
-                continue
-        else:
-            trigger = event_id(desc)
-            if trigger is not None:
-                event_id_[desc] = trigger
-            else:
-                dropped.append(desc)
+    event_sel, event_id_ = _select_annotations_based_on_description(
+                    annotations.description, event_id=event_id, regexp=regexp)
 
-    event_sel = [ii for ii, kk in enumerate(annotations.description)
-                 if kk in event_id_]
+    events = np.empty((0,2), dtype=int)
+    iterator = list(zip(annotations.onset[event_sel],
+                    annotations.duration[event_sel],
+                    annotations.description[event_sel]))
 
-    if len(event_sel) == 0 and regexp is not None:
-        raise ValueError('Could not find any of the events you specified.')
-
-    values = [event_id_[kk] for kk in
-              annotations.description[event_sel]]
-    previous_value = np.zeros(len(event_sel))
-    inds = inds[event_sel]
-    events = np.c_[inds, previous_value, values].astype(int)
-
-    logger.info('Used Annotations descriptions: %s' %
-                (list(event_id_.keys()),))
+    for onset, duration, description in iterator:
```

Comment:
I would like to add slicing of the annotations so that this can be:
`for onset, duration, description in raw.annotations[event_sel]:`

Reply:
but this can be done afterwards

Reply:
Yes a `__getitem__` that returns a copy and a `__iter__` method (or whatever the correct Py3k one is named) would be good at some point